server/router_utils.go

```diff
@@ -206,3 +206,34 @@ func getRuleAndErrorKeyFromRuleID(ruleIDWithErrorKey string) (
 	errorKey = types.ErrorKey(splitedRuleID[1])
 	return
 }
+
+// ReadClusterMapFromBody retrieves a list of clusters from the request body and
+// generates a map of empty structs for fast access. If it's not possible,
+// it writes an http error to the writer and returns false
+func ReadClusterMapFromBody(writer http.ResponseWriter, request *http.Request) (
+	clusterMap map[types.ClusterName]struct{},
+	successful bool,
+) {
+	// check if there's any body provided in the request sent by client
+	if request.ContentLength <= 0 {
+		err := &NoBodyError{}
+		handleServerError(writer, err)
+		return
+	}
+
+	// decode cluster list from body
+	var clustersInRequest []types.ClusterName
+	err := json.NewDecoder(request.Body).Decode(&clustersInRequest)
+	if err != nil {
+		handleServerError(writer, err)
+		return
+	}
+
```

Comment:
"by making a POST request, the cluster filtering is enabled, meaning that if a valid empty list is provided {[]}, the handler will filter every result out, returning an empty results set."

What about, if the length of the received list is 0, returning the empty results set directly? We'd save some processing that way.